backend/workers/lib/vod_converter/kitti.py

```diff
@@ -0,0 +1,178 @@
+        try:
+            detections_fpath = f"{root}/labels/{image_id}.txt"
+            detections = self._get_detections(detections_fpath, image_id)
+            detections = [det for det in detections if det['left'] < det['right'] and det['top'] < det['bottom']]
+            image_path = f"{root}/images/{image_id}.{image_ext}"
+            image_width, image_height = _image_dimensions(image_path)
+            return {
+                'image': {
+                    'id': image_id,
+                    'dataset_id': None,
+                    'path': image_path,
+                    'segmented_path': None,
+                    'width': image_width,
+                    'height': image_height,
+                    'file_name': image_id + '.' + image_ext
+                },
+                'detections': detections
+            }
+        except Exception as e:
+            print(e)
+
+    def _get_detections(self, detections_fpath, image_id):
+        detections = []
+        with open(detections_fpath) as f:
+            f_csv = csv.reader(f, delimiter=' ')
+            for row in f_csv:
+                if len(row) == 0:
+                    continue
+                try:
+                    x1, y1, x2, y2 = map(float, row[4:8])
+                    label = row[0]
+                    detections.append({
+                        'id': image_id,
+                        'image_id': image_id,
+                        'label': label,
+                        'left': x1,
+                        'right': x2,
+                        'top': y1,
+                        'bottom': y2,
+                        'area': None,
+                        'segmentation': None,
+                        'isbbox': True,
+                        'iscrowd': False,
+                        'keypoints': []
+                    })
+                except ValueError as ve:
+                    print(row)
+        return detections
+
+
+def _image_dimensions(path):
+    with Image.open(path) as image:
+        return image.width, image.height
+
+
+DEFAULT_TRUNCATED = 0.0  # 0% truncated
+DEFAULT_OCCLUDED = 0  # fully visible
+
+
+class KITTIEgestor(Egestor):
+
+    def expected_labels(self):
+        return output_labels
+
+    def egest(self, *, image_detections, root, folder_names):
+        images_dir = f"{root}/images"
+        os.makedirs(images_dir, exist_ok=True)
+        labels_dir = f"{root}/labels"
+        os.makedirs(labels_dir, exist_ok=True)
+
+        id_file = f"{root}/train.txt"
+
+        for image_detection in image_detections:
+            image = image_detection['image']
+            image_id = image['id']
+            print(image_id)
+            src_extension = image['path'].split('.')[-1]
+            try:
+                shutil.copyfile(image['path'], f"{images_dir}/{image_id}.{src_extension}")
+            except FileNotFoundError as e:
+                print(e)
```

Comment:
use logger

Reply:
sys.stdout was temporarily redirected to task.info to omit the need for replacing all prints in new scripts